dotcom-rendering/.storybook/main.js

```diff
@@ -0,0 +1,140 @@
+					minify: false,
+					minifyWhitespace: true,
+					minifyIdentifiers: false,
+					minifySyntax: true,
+				},
+			},
+		},
+	],
+	webpackFinal: async (config) => {
+		// Get project specific webpack options
+		config = webpackConfig(config);
+
+		// Global options for webpack
+		config.resolve.extensions.push('.ts', '.tsx');
+
+		// Required as otherwise 'process' will not be defined when included on its own (without .env)
+		// e.g process?.env?.SOME_VAR
+		config.plugins.push(
+			new webpack.DefinePlugin({
+				process: '{}',
+			}),
+		);
+
+		return config;
+	},
+	env: (config) => ({
+		...config,
+		// Github sets a CI env var for all actions but this isn't being picked up by Storybook
+		// See: https://storybook.js.org/docs/react/configure/environment-variables
+		CI: 'true',
+	}),
+};
+
+const webpackConfig = (config) => {
+	const rules = config.module.rules;
+
+	// Mock JSDOM for storybook - it relies on native node.js packages
+	// Allows us to use enhancers in stories for better testing of components & full articles
+	config.resolve.alias.jsdom$ = path.resolve(__dirname, './mocks/jsdom.js');
+
+	// log4js tries to call "fs" in storybook -- we can ignore it
+	config.resolve.alias[
+		path.resolve(__dirname, '../src/server/lib/logging.ts')
+	] = path.resolve(__dirname, './mocks/log4js.js');
+
+	// SecureSignup uses @emotion/cache and @emotion/server - can't be used in storybook
+	config.resolve.alias[
+		path.resolve(__dirname, '../src/web/components/SecureSignup.tsx')
+	] = path.resolve(__dirname, '../__mocks__/SecureSignupMock.tsx');
+
+	// Support typescript in Storybook
+	// https://storybook.js.org/docs/configurations/typescript-config/
+	rules.push({
+		test: /\.[jt]sx?|mjs$/,
+		include: [
+			path.resolve(__dirname, '../'),
+			path.resolve(__dirname, '../../common-rendering'),
+		],
+		exclude: require('../scripts/webpack/webpack.config.browser')
+			.babelExclude,
+		use: [
+			{
+				loader: 'babel-loader',
+				options: {
+					presets: [
+						'@babel/preset-react',
+						[
+							'@babel/preset-env',
+							{
+								bugfixes: true,
+								targets: {
+									esmodules: true,
+								},
+							},
+						],
+					],
+				},
+			},
+			{
+				loader: 'ts-loader',
+				options: {
+					configFile: 'dotcom-rendering/tsconfig.build.json',
+					transpileOnly: true,
+				},
+			},
+		],
+	});
+
+	// modify storybook's file-loader rule to avoid conflicts with our svg
+	// https://stackoverflow.com/questions/54292667/react-storybook-svg-failed-to-execute-createelement-on-document
+	const fileLoaderRule = rules.find((rule) => rule.test.test('.svg'));
+	fileLoaderRule.exclude = /\.svg$/;
+	rules.push({
+		test: /\.svg$/,
+		use: ['desvg-loader/react', 'svg-loader'],
+	});
+
+	config.resolve.modules = [
+		...((config && config.resolve && config.resolve.modules) || []),
+		path.resolve(__dirname, '../../common-rendering/src'),
+	];
+
+	config.resolve.alias = {
+		...config.resolve.alias,
+	};
+
+	return config;
+};
```

Comment:
I’m assuming this has not changed from what it used to be.

Reply:
this is the same except for updated paths